cmd/juju/machine/upgradeseries.go

```diff
@@ -215,6 +231,83 @@ func (c *upgradeSeriesCommand) UpgradeSeriesPrepare(ctx *cmd.Context) error {
 		return errors.Trace(err)
 	}
 
+	err = c.handleNotifications(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	m := UpgradeSeriesPrepareFinishedMessage + "\n"
+	fmt.Fprintf(ctx.Stdout, m, c.machineNumber)
+
+	return nil
+}
+
+func (c *upgradeSeriesCommand) handleNotifications(ctx *cmd.Context) error {
+	if c.plan.Work == nil {
+		c.plan = catacomb.Plan{
+			Site: &c.catacomb,
+			Work: c.displayNotifications(ctx),
+		}
+	}
+	err := catacomb.Invoke(c.plan)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// Here we wait for the loop to finish by waiting for the catacomb's
+	// worker to finish.
+	err = c.catacomb.Wait()
+	if err != nil {
+		if err.Error() == "watcher has been stopped (stopped)" {
```

Comment:
OMG, I am checking against a string! No good. What should we do here instead. My ideas:

1. Somehow get the catacomb to error with something better ( have the watcher api return a type or value to be matched against.
2. Stop the worker loop in another fashion besides than killing the watcher.